### PR TITLE
Search transform

### PR DESCRIPTION
**Pull request by jhidalgocarrio**
This PR includes the search algorithm to get any Transformation in the Graph. The tricky part wrt a "classical" graph is that transformations do not hold the commutative property. It uses the BFS which behaves like a Dijkstra with no weight in edges. It returns the shortest part between two given Frames. Uni test available and tested. Other minor things included in this PR is to have frameId and frame names to the same value.

### CODE REVIEW

File: src/TransformGraphVisitors.hpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                    try
+                    {
+                        parent_vertex = parent->at(parent_vertex);
+                    }catch (const std::out_of_range& oor)
+                    {
+                        parent_vertex = Graph::null_vertex();
+                    }
+                }
+
+                throw FoundFrameException(g[v].frame.name);
```
Comment (arneboe) [inline src/TransformGraphVisitors.hpp]: Is this the normal way to return a value from boost visitors?
It seems a bit strange to use an exception.

Comment (jhidalgocarrio) [inline src/TransformGraphVisitors.hpp]: The search algorithm does not stop until it visits all the vertices in the graph.
This is the only manner I found and also recommended in some forums.

**Comment by arneboe** [comment]
Oops, I caused some conflicts :)
I'll fix them and merge this PR if nobody has any more problems with it?

**Comment by jhidalgocarrio** [comment]
+1

**Pull request merged**